internal/interpreter/json.rs

```diff
@@ -0,0 +1,284 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+//! This module contains the code serialize and desrialize `Value`s to JSON
+
+use std::collections::HashMap;
+
+use i_slint_compiler::langtype;
+use i_slint_core::{
+    graphics::Image,
+    model::{Model, ModelRc},
+    Brush, Color, SharedString, SharedVector,
+};
+
+use crate::Value;
+
+/// Extension trait, adding JSON serialization methods
+pub trait JsonExt
+where
+    Self: Sized,
```

Comment:
What is this bound for?

Reply:
So that `Result<Self, String>` is `Sized`.